src/firecracker/src/api_server_adapter.rs

```diff
@@ -146,34 +144,36 @@ pub(crate) fn run_with_api(
         .expect("Missing seccomp filter for API thread.");
 
     // Start the separate API thread.
-    let api_thread = thread::Builder::new()
-        .name("fc_api".to_owned())
-        .spawn(move || {
-            match ApiServer::new(to_vmm, from_vmm, to_vmm_event_fd).bind_and_run(
-                &api_bind_path,
-                process_time_reporter,
-                &api_seccomp_filter,
-                api_payload_limit,
-                socket_ready_sender,
-            ) {
-                Ok(_) => (),
-                Err(api_server::Error::ServerCreation(ServerError::IOError(inner)))
-                    if inner.kind() == std::io::ErrorKind::AddrInUse =>
-                {
-                    let sock_path = api_bind_path.display().to_string();
-                    error!(
-                        "Failed to open the API socket at: {sock_path}. Check that it is not \
-                         already used."
-                    );
-                    std::process::exit(vmm::FcExitCode::GenericError as i32);
-                }
-                Err(api_server::Error::ServerCreation(err)) => {
-                    error!("Failed to bind and run the HTTP server: {err}");
-                    std::process::exit(vmm::FcExitCode::GenericError as i32);
-                }
+
+    let api_thread = tokio::task::spawn(async move {
+        match api_server::bind_and_run(
```

Comment:
fyi: this will run the API server as an async task on the same (now tokio) thread as emulation (which is with this PR also an async tokio task, rather than dedicated thread).

this could help with VMs density with potentially fewer VM threads on the host, but could also have unintended consequences on emulation latency (thus overall VM performance)

an important question to ask here is: what type of HTTP traffic can there be on the API server? is the api server privileged, permissioned, permissionless? basically whether it's ultimately ok to share the thread or not - if API is privileged and traffic on it is inconsequential, then this change should be perfectly ok.